modules/apps/dynamic-data-mapping/dynamic-data-mapping-test/src/testIntegration/java/com/liferay/dynamic/data/mapping/internal/upgrade/v1_0_1/test/DDMFormPortletPreferencesUpgradeProcessTest.java

```diff
@@ -0,0 +1,139 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v1_0_1.test;
+
+import com.liferay.arquillian.extension.junit.bridge.junit.Arquillian;
+import com.liferay.dynamic.data.mapping.constants.DDMPortletKeys;
+import com.liferay.dynamic.data.mapping.model.DDMFormInstance;
+import com.liferay.dynamic.data.mapping.model.DDMStructure;
+import com.liferay.dynamic.data.mapping.service.DDMStructureLocalService;
+import com.liferay.dynamic.data.mapping.test.util.DDMFormInstanceTestUtil;
+import com.liferay.layout.test.util.LayoutTestUtil;
+import com.liferay.portal.kernel.cache.MultiVMPool;
+import com.liferay.portal.kernel.model.Group;
+import com.liferay.portal.kernel.model.Layout;
+import com.liferay.portal.kernel.service.PortletPreferencesLocalService;
+import com.liferay.portal.kernel.test.util.GroupTestUtil;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortletKeys;
+import com.liferay.portal.test.log.LogCapture;
+import com.liferay.portal.test.log.LoggerTestUtil;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.test.rule.LiferayIntegrationTestRule;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.Map;
+
+import javax.portlet.PortletPreferences;
+
+import org.junit.Assert;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+/**
+ * @author Paulo Albuquerque
+ */
+@RunWith(Arquillian.class)
+public class DDMFormPortletPreferencesUpgradeProcessTest {
+
+	@ClassRule
+	@Rule
+	public static final LiferayIntegrationTestRule liferayIntegrationTestRule =
+		new LiferayIntegrationTestRule();
+
+	@Test
+	public void testUpgrade() throws Exception {
+		Group group = GroupTestUtil.addGroup();
+
+		DDMFormInstance ddmFormInstance =
+			DDMFormInstanceTestUtil.addDDMFormInstance(
+				GroupTestUtil.addGroup(), TestPropsValues.getUserId());
+
+		Layout layout = LayoutTestUtil.addTypeContentLayout(group);
+
+		String portletId = LayoutTestUtil.addPortletToLayout(
+			layout, DDMPortletKeys.DYNAMIC_DATA_MAPPING_FORM);
+
+		PortletPreferences portletPreferences =
+			LayoutTestUtil.getPortletPreferences(layout, portletId);
+
+		portletPreferences.setValue(
+			"formInstanceId",
+			String.valueOf(ddmFormInstance.getFormInstanceId()));
+		portletPreferences.setValue(
+			"groupId", String.valueOf(group.getGroupId()));
+
+		_portletPreferencesLocalService.updatePreferences(
+			PortletKeys.PREFS_OWNER_ID_DEFAULT,
+			PortletKeys.PREFS_OWNER_TYPE_LAYOUT, layout.getPlid(), portletId,
+			portletPreferences);
+
+		portletPreferences = LayoutTestUtil.getPortletPreferences(
+			layout, portletId);
```

Comment:
@p-albuquerque why do you need to do a `getPortletPreferences` after the update? 

Can't you just do:

```
portletPreferences = _portletPreferencesLocalService.updatePreferences(...)
```
?

Reply:
@carolmariaabb  I just want to ensure the values were stored in the database before the upgrade, but maybe it's make no sense. I will do as you said.

Reply:
Actually, now I noticed the PortletPreference returned by `updatePreferences` and `getPortletPreferences` are different type